Index: d03/Defdemo3.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- d03/Defdemo3.py	(date 1594728780309)
+++ d03/Defdemo3.py	(date 1594728780309)
@@ -0,0 +1,13 @@
+#有回傳值得方法
+def getBMI(h, w):
+    bmi = w/((h/100)**2)
+    return bmi
+
+bmi = getBMI(170, 60)
+print("%.2f" %bmi)
+
+bmi = getBMI(180, 70)
+print("%.3f" %bmi)
+
+bmi = getBMI(160, 50)
+print("%.4f" %bmi)
\ No newline at end of file
Index: d03/if else Demo.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- d03/if else Demo.py	(date 1594725182612)
+++ d03/if else Demo.py	(date 1594725182612)
@@ -0,0 +1,11 @@
+def printBMI(h,w):
+    bmi = w/((h/100)**2)
+    if(bmi > 23):
+        result = "過重"
+    elif(bmi < 18):
+        result = "過輕"
+    else:
+        result = "正常"
+    print("身高: %.1f 體重: %.1f BMI: %.2f (%s)" % (h,w,bmi,result))
+
+printBMI(170.5, 60.3)
Index: d03/if else Demo2.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- d03/if else Demo2.py	(date 1594725890747)
+++ d03/if else Demo2.py	(date 1594725890747)
@@ -0,0 +1,12 @@
+def getBMI(h,w): # 建 getBMI(回傳值=bmi, result, h, w)
+    bmi = w/((h/100)**2)
+    if(bmi > 23):
+        result = "過重"
+    elif(bmi < 18):
+        result = "過輕"
+    else:
+        result = "正常"
+    return bmi, result, h, w  #return = 回傳
+
+bmi, result, h, w = getBMI(170.5, 60.5)
+print("身高: %.1f 體重: %.1f BMI: %.2f (%s)" % (h,w,bmi,result))
\ No newline at end of file
Index: d03/if else Demo3.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- d03/if else Demo3.py	(date 1594728011333)
+++ d03/if else Demo3.py	(date 1594728011333)
@@ -0,0 +1,30 @@
+# 撰寫 BMI計算系統
+def calcBMI():  #副程式
+    h = float(input('你的身高(cm): '))
+    w = float(input('你的體重(kg): '))
+    bmi = w/((h/100)**2)
+    result = "過重" if (bmi > 23) else "過輕" if (bmi < 18) else ("正常")
+    print("身高: %.1f 體重: %.1f BMI: %.2f (%s)" % (h, w, bmi, result))
+
+
+def menu():   #副程式
+    print(" ")
+    print(" ")
+    print("BMI 計算系統")
+    print("-----------")
+    print("1. 輸入身高與體重")
+    print("2. 離開系統")
+    id = int(input('請選擇 : '))
+    if id == 1:
+        print('您選擇的是 1')
+        calcBMI()
+        input('按下任意鍵繼續')
+        menu()
+    elif id == 2:
+        print('您選擇的是 2')
+        print('謝謝您的使用')
+    else:
+        print('選擇錯誤')
+        menu()
+
+menu()   #主程式
\ No newline at end of file
Index: d03/fibonacci.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- d03/fibonacci.py	(date 1594728780359)
+++ d03/fibonacci.py	(date 1594728780359)
@@ -0,0 +1,12 @@
+# 求解費式數列第30項
+def fib(n):
+    if n == 0:
+        return 0
+    elif n == 1:
+        return 1
+    else:
+        return fib(n-1) + fib(n-2)
+
+print(fib(30))
+
+# 0,1,1,2,3,5,8,13,21,34......
\ No newline at end of file
